app/src/main/kotlin/com/aliucord/manager/ui/viewmodel/InstallViewModel.kt

```diff
@@ -54,55 +60,96 @@ class InstallViewModel(
 
                 log += "Downloading drawables APK... "
                 downloadManager.downloadSplit(supportedVersion, "config.xxhdpi")
                 log += "Done\n"
-            }
+            }.copyTo(externalCacheDir.resolve("patched").resolve("config.xxhdpi-${supportedVersion}.apk"), true)
+
+            val (hermesLibrary, cppRuntimeLibrary) = run {
+                val latestHermesRelease = githubRepository
+                    .getHermesReleases().reduce { current, new ->
+                        if (Instant.parse(current.createdAt).isBefore(Instant.parse(new.createdAt))) new else current
+                    }
+                val hermesFile = externalCacheDir.resolve("hermes-release-${latestHermesRelease.tagName}.aar")
+                val cppRuntimeFile = externalCacheDir.resolve("hermes-cppruntime-release-${latestHermesRelease.tagName}.aar")
+                if (
+                    hermesFile.exists()
+                    && cppRuntimeFile.exists()
+                ) arrayOf(hermesFile, cppRuntimeFile).also { log += "Using cached patched libraries\n" }
+                else {
+                    log += "Downloading patched libraries... "
+
+                    arrayOf(
+                        downloadManager.download(
+                            url = latestHermesRelease.assets.find { it.name == "hermes-release.aar" }!!.browserDownloadUrl,
+                            fileName = hermesFile.name
+                        ),
+                        downloadManager.download(
+                            url = latestHermesRelease.assets.find { it.name == "hermes-cppruntime-release.aar" }!!.browserDownloadUrl,
+                            fileName = cppRuntimeFile.name
+                        )
+                    ).also { log += "Done\n" }
+                }
+            }.map { it.copyTo(externalCacheDir.resolve("patched").resolve(it.name), true) }
+
+            val latestAliucordNativeRelease = githubRepository
+                .getAliucordNativeReleases().reduce { current, new ->
+                    if (Instant.parse(current.createdAt).isBefore(Instant.parse(new.createdAt))) new else current
+                }
 
-            val apks = arrayOf(baseApk, libsApk, localeApk, xxhdpiApk)
+            val aliucordDexFile = externalCacheDir.resolve("classes-${latestAliucordNativeRelease.tagName}.dex").also { file ->
+                if (file.exists()) return@also run { log += "Using cached Aliucord dex file\n" }
 
-            log += "Patching manifest...\n"
+                log += "Downloading Aliucord dex file... "
+                downloadManager.download(
+                    url = latestAliucordNativeRelease.assets.find { it.name == "classes.dex" }!!.browserDownloadUrl,
+                    fileName = "classes-${latestAliucordNativeRelease.tagName}.dex"
+                ).also { log += "Done\n" }
+            }.copyTo(externalCacheDir.resolve("patched").resolve("classes-${latestAliucordNativeRelease.tagName}.dex"), true)
```

Comment:
What happened to the beautifully formatted code like below I will explode from inline ifs